autolab/Autolab
# Rails upgrade

## Description
This PR updates rails to 6.0.0, updates relevant dependencies, and fixes deprecation bugs.

## Motivation and Context
Rails 6 has been out forever. Lots of dependency issues in the past.

## How Has This Been Tested?
In development environment:
To set up,
1. Delete `Gemfile.lock` and do `bundle install`. You might need to upgrade rails first. Follow the instructions in your terminal.
2. Re-migrate the database (sqlite3 has been updated).
3. Follow the internal testing sheets in Google Drive.

**I have done this part. I need someone to test for production environment.**

In production environment:
1. Use docker-compose to re-build the images.
2. Follow the internal testing sheets in Google Drive.

Known bugs that exist on current master branch: #1417  #1416 
Make sure to check back on existing bug reports if you find any bug while testing.

## Types of changes


## Checklist:


## Other issues / help required




If unsure, feel free to submit first and we'll help you along.
